msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md

```diff
@@ -0,0 +1,79 @@
+
+|ad-hoc proactive message|dialog-based proactive message|
+|-----------------------------|------------------------------------|
+|The bot interjects a message without interrupting the conversation flow. | The bot creates a new dialog thread, takes control of a conversation, delivers the proactive message, closes, and returns control to the previous dialog.|
+
+## What is proactive app installation in Teams?
+
+Before your bot can proactively message a user, it needs to be installed either as a personal app, or in a team where the user is a member. At times,  you may need to proactively message users that have _not_ installed or previously interacted with your app. For instance, the need to message vital information to everyone in your organization. For that scenario you can use the Microsoft Graph API to proactively install your bot for your users and cache the necessary values received by your bot at install.
+
+Microsoft Graph [teamsAppInstallationresource type](/graph/api/resources/teamsappinstallation?view=graph-rest-1.0) permissions allow your bot to manage its installation lifecycle for all user (personal ) or team (group) scopes within the Microsoft Teams platform:
+
+|Application permission | Description|
+|------------------|---------------------|
+|`TeamsAppInstallation.ReadWriteSelfForUser.All`|Allows a Teams app to read, install, upgrade, and uninstall itself to any **user**, without a prior sign in or use.|
+|`TeamsAppInstallation.ReadWriteSelfForTeam.All`|Allows a Teams app to read, install, upgrade, and uninstall itself in any **team**, without a prior sign in or use.|
+
+>[!NOTE]
+>
+> * Your bot must have _application permissions_ not _user delegated permissions_ because the installation is not for yourself but for others.
+>
+> * An Azure AD tenant administrator must [explicitly grant permissions to an application](/graph/security-authorization#grant-permissions-to-an-application). After an application is granted permissions, _all_ members of the Azure AD tenant will gain the granted permissions.
+
+## Enable proactive app installation and messaging
+
+ > [!IMPORTANT]
+> Microsoft Graph will only install apps published in your organization's [app catalog](../../concepts/deploy-and-publish/overview.md#publish-to-your-organizations-app-catalog) or [AppSource](https://appsource.microsoft.com/).
+
+### ✔ Determine whether your bot is currently installed for all intended users
+
+**HTTP GET** request:
+
+```http
+GET /users/{user-id}/teamwork/installedApps?$expand=teamsAppDefinition&$filter=teamsAppDefinition/teamsAppId eq '{teamsAppid}'
+```
+
+### ✔ Install your app for subsequent users
+
+**HTTP POST** request:
+
+```http
+POST /users/{user-id}/teamwork/installedApps
```

Comment:
For each of these calls, let's provide a link to the corresponding Graph reference page